components/core/spotlight.tsx

```diff
@@ -0,0 +1,71 @@
+'use client';
+import React, { useRef, useState, useCallback } from 'react';
+import { motion, useSpring, useTransform } from 'framer-motion';
+import { cn } from '@/lib/utils';
+
+type SpotlightProps = {
+  children: React.ReactNode;
+  className?: string;
+  spotlightSize?: number;
+  color?: string;
+  opacity?: number;
+  blur?: number;
+  transition?: {
```

Comment:
you can use `SpringOptions` type from framer-motion. And rename the prop `springOptions`